aws/resource_aws_dms_replication_task.go

```diff
@@ -328,3 +333,29 @@ func resourceAwsDmsReplicationTaskStateRefreshFunc(
 		return v, *v.ReplicationTasks[0].Status, nil
 	}
 }
+
+func cleanReadOnlyReplicationTaskSettings(settings string) (*string, error) {
+	var settingsData map[string]interface{}
+	if err := json.Unmarshal([]byte(settings), &settingsData); err != nil {
+		return nil, err
+	}
+
+	controlTablesSettings, ok := settingsData["ControlTablesSettings"].(map[string]interface{})
+	if ok {
+		delete(controlTablesSettings, "historyTimeslotInMinutes")
+	}
+
+	logging, ok := settingsData["Logging"].(map[string]interface{})
+	if ok {
+		delete(logging, "CloudWatchLogGroup")
+		delete(logging, "CloudWatchLogStream")
```

Comment:
If you define these in your task settings, DMS API will return an error that they are read only.